### ISSUE TITLE
Document highlight for symbols (field names, model names, external ids,...)

### ISSUE THREAD

**Issue by trinhanhngoc** [odoo-ide/vscode-odoo#39]

**Comment by gustitammam** [comment]
Could this issue be related to **undefined methods being referenced in field attributes**, such as a compute method that hasn't been defined yet? 

Currently, in my VSCode, there's no warning or highlight indicator for such cases, making it difficult to identify undefined methods being referenced. When can we expect this issue to be resolved?

**Comment by trinhanhngoc** [comment]
@gustitammam ,

No, the document highlight feature is about highlighting all references to the symbol under the cursor. Yours is another problem. Can you take a screenshot to demonstrate the problem?

**Comment by gustitammam** [comment]
Thank you for the clarification! I was searching for an existing issue that might match the problem I’m facing because I want to avoid creating a duplicate issue. I thought this issue might be related, so I wanted to confirm.

The expected behavior is that if a method referenced in the compute attribute of a field is not defined or doesn't exist, there should be a red highlight or a similar warning—like the functionality available in PyCharm.

**PyCharm:**
![image](https://github.com/user-attachments/assets/ab7b573a-fd8f-4e04-85e8-e798d698ed9c)


Currently, no such indicator appears in VSCode.

**VSCode:**
![image](https://github.com/user-attachments/assets/8b7a6391-4adf-4d87-8b53-a7722d28855d)

**Comment by trinhanhngoc** [comment]
@gustitammam ,

I just tested, everything is working fine. Are you sure the Odoo IDE extension is enabled?

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/cda88afa-ad11-4e04-b65f-f2d2dddfb26e" />

**Comment by gustitammam** [comment]
That's strange because on my side it doesn't seem to work as expected.

Do you think I should open a separate issue for this?

I've changed the code in the Odoo source to match yours, but I still don't see any warning or indicator when an invalid method is referenced in the compute attribute of a field. I've attached a screenshot to demonstrate this.

For additional context, here's my setup:

📌 **VSCode version:**
```
Version: 1.96.2
Commit: fabdb6a30b49f79a7aba0f2ad9df9b399473380f
Date: 2024-12-19T10:22:47.216Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Linux x64 6.12.5-200.fc41.x86_64
```

📌 **Extension version:**
```
Identifier: trinhanhngoc.vscode-odoo
Version: 0.36.4
```
📌 **Installed inside a devcontainer**

📌 **Using a multi-root workspace (configuration attached)**
```json
{
  "folders": [
    {
      "path": "."
    },
    {
      "path": "/oo"
    }
  ],
  "settings": {
    "remote.autoForwardPorts": false,
    "remote.restoreForwardedPorts": false,
    "files.exclude": {
      "**/.git": true,
      "**/.svn": true,
      "**/.hg": true,
      "**/CVS": true,
      "**/.DS_Store": true,
      "**/Thumbs.db": true,
      "**/__pycache__": true,
      "**/.idea": true
    },
    "search.exclude": {
      "workdir/data/**/*": true
    },
    "git.repositoryScanMaxDepth": 3,
    "git.allowForcePush": true,
    "python.defaultInterpreterPath": "/usr/local/bin/python",
    "python.createEnvironment.trigger": "off",
    "python.languageServer": "None",
    "python.analysis.stubPath": "",
    "autopep8.args": [
      "--max-line-length=119"
    ],
    "xml.format.maxLineWidth": 120,
    "xml.format.spaceBeforeEmptyCloseTag": false,
    "editor.rulers": [
      {
        "column": 120
      }
    ],
    "[python]": {
      "editor.defaultFormatter": "ms-python.autopep8",
      "editor.rulers": [
        {
          "column": 79
        },
        {
          "column": 119
        }
      ]
    },
    "debug.showInStatusBar": "always"
  }
}
```

![image](https://github.com/user-attachments/assets/01247ad6-41ca-450d-922b-e07634e19d9d)

The extension log also confirms that the file is being analyzed, so it seems the extension is functioning correctly. Let me know if there's anything else you'd like me to check.

**Comment by trinhanhngoc** [comment]
@gustitammam ,

Are other features of the Odoo IDE extension still working (code completion for method names in `compute='...'`, model names in `env['...']`,...)?

**Comment by gustitammam** [comment]
> @gustitammam ,
> 
> Are other features of the Odoo IDE extension still working (code completion for method names in `compute='...'`, model names in `env['...']`,...)?

Yes, everything else is working perfectly.

**Code completion for method names:**
This is working in both the Odoo source code and custom source code:

![image](https://github.com/user-attachments/assets/ca67ab87-7f7d-42c5-afc5-da962fe06781)

**Code completion for model names:**
This is also working in both the Odoo source code and custom source code:

![image](https://github.com/user-attachments/assets/b47f16e3-4fb7-48b8-b1f8-f6a12be83a55)


Additionally, 'Go to Definition' is functioning correctly as well. Everything else seems to be working as expected.

**Comment by trinhanhngoc** [comment]
@gustitammam ,

What about your **Type Checking Mode** setting?

<img width="1084" alt="image" src="https://github.com/user-attachments/assets/faefc47e-8e10-449f-a624-065bf2eb8b75" />

**Comment by gustitammam** [comment]
Apologies for the late reply. The **Type Checking Mode** setting in my VSCode was set to **Off**.

I hadn't noticed this setting before, and it seems the default is Off. After changing it to **"basic"**, the feature is now working as expected. Thank you for pointing that out!

![image](https://github.com/user-attachments/assets/aeac51d3-8e3a-434b-a3c5-248caecfb758)
